ltj/settings.py

```diff
@@ -72,7 +72,7 @@
     'rest_framework',
     'rest_framework_gis',
     'nature',
-    'import',
+    'imports',
```

Comment:
I think it's related to my coding habits that I refactor quite often when doing relative big feature, and quite often there're some other things have been done before I realize something can be improved/refactored. This (renaming especially) make the rewriting commit history very difficult as the later commits relies on the name done in previous commits. I know there're discussion out there that we should make the commit history clean, but I personally prefer to try keeping the commit history as untouched as it reflects the spirit of software craftsmanship, we rarely get everything right in the first place but rather it's a gradual process while our understanding the of problem becomes better and better.

I just tried to rebase the commits of those renamings, and as I expected, it blows up all the commits and require fixing merge conflicts all the way up to the latest commits. So if you do not have a strong opinion on this, I'd rather keep it as is, or I think I can quickly merge all the commits into a single one (resolve all merge conflicts in the commit tree would be too troublesome).

Reply:
@frwickst I merged the consecutive commits from the first one to the one that renames the app. But left other renaming commits after that as is because there are relative big changes between them.

Reply:
I do kinda have a strong opinion here.

I think that more often then not the reason for not having things merged like in this case is just people not wanting to put in the work to do so. I'm not saying that is the case here, and I have seen several people state the same thing that they want everything to be "as they developed". That however to me serves no purpose, when there are several people working on projects, all working in a different manner and then you get a bunch of commits that goes back and forth and the history does not make sense in the end. Consistency is thrown out the window this happens and the git history becomes a mess.

Another reason for keeping the history in check is that if done really well, all commits should be testable and pass all tests. At least most of the time. This way `git bisect` can be used properly for instance. I know that this is not always the case and it is really hard to strive for such a codebase where all commits are testable and that is why I've never put that restriction on myself.

Merging everything into a single commit is not really a good way to do it either since there are clearly several things happening here and only having one commit will hurt the git history as well.

I took a few minutes to try to refactor this myself and got to this point:
https://github.com/frwickst/ltj/commits/shp-import
Since I did not write the original code I did not put a lot of time into checking everything out and I think it is best that you have a quick look that everything is still intact, but all should be good.

Reply:
I already did so with the previous commits, but would leave the ShpImport -> ShapefileImport as separate commit as it does not only change the model name, but also change all other related classes.

> Another reason for keeping the history in check is that if done really well, all commits should be testable and pass all tests

I agree with that and think that's actually a more important thing to keep with, and usually would run full suit tests after each commit.  But that's actually another reason that I'm reluctant to do rebasing and rewrite commit history because it kind of making it difficult to verify that each commit still pass all the tests after reorder or squash commits.